Debug-backend-tests.md

```diff
@@ -0,0 +1,249 @@
+## Introduction
+
+## Interpret log output
+
+When your backend tests fail, look near the end of the log for output that looks like this:
+
+```text
+======================================================================
+FAIL: test_failed_api_call_logs_the_exception (jobs.jobs_manager_test.RefreshStateOfBeamJobRunModelTests)
+----------------------------------------------------------------------
+Traceback (most recent call last):
+  File "/home/runner/work/oppia/oppia/jobs/jobs_manager_test.py", line 158, in test_failed_api_call_logs_the_exception
+    lambda: jobs_manager.refresh_state_of_beam_job_run_model(
+  File "/home/runner/work/oppia/oppia/core/tests/test_utils.py", line 1306, in assertRaisesRegexp
+    expected_exception, expected_regex, *args, **kwargs)
+AssertionError: Exception not raised by <lambda>
+
+---------------------------------------------------------------------
+```
+
+This is the part of the log that you should pay attention to. It tells you which test failed and why the failure occurred. Here's how to read it:
+
+* At the top, the line beginning with `FAIL` tells you which test failed. In this case, it's the `test_failed_api_call_logs_the_exception` test in the `jobs.jobs_manager_test.RefreshStateOfBeamJobRunModelTests` class. Note that this line can also start with `ERROR`. `FAIL` indicates that an assertion failed, while `ERROR` indicates that an exception (other than `AssertionError`) was raised unexpectedly.
+* Next is the traceback. This shows you the state of the stack when the error was raised. In this example:
+
+  * The `AssertionError` was raised by line 1306 of `test_utils.py` in the `assertRaisesRegexp()` function.
+  * That `assertRaisesRegexp()` function was called on line 158 of `jobs_manager_test.py` in the `test_failed_api_call_logs_the_exception` function. Since this is the function defining our test, it's the end of the trace.
```

Comment:
```suggestion
  * That `assertRaisesRegexp()` function was called on line 158 of `jobs_manager_test.py` in the `test_failed_api_call_logs_the_exception()` function. Since this is the function defining our test, it's the end of the trace.
```